### ISSUE TITLE
[2.1] - ROADMAP

### ISSUE THREAD

**Issue by dagatsoin** [Telerik-Verified-Plugins/Mapbox#71]
Project Collaborators:
@EddyVerbruggen @cusspvz @anothar @dagatsoin

Refactoring:

 [Issue #] Base native events trough an initial event listener callback for each map instance
 [Issue #] Use latest ES methods on JS file
Features:

 [Feature #70] Add support for ionic-native
Maintenance:

 [Issue #70] Wrap JS API in Ionic Typescript wrapper.

**Comment by ansarikhurshid786** [comment]
how to use this plugin with ionic 3?

**Comment by risinghero** [comment]
@ansarikhurshid786 please stop spamming!!! I have 6 email notifications all with the same your question!

**Comment by ansarikhurshid786** [comment]
@risinghero No, I am not spamming. You mentioned somewhere that this plugin work with ionic 1, 2 or any.
